checkov/terraform/checks/resource/azure/AKSLoggingEnabled.py

```diff
@@ -1,17 +1,26 @@
-from checkov.common.models.enums import CheckCategories
-from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceValueCheck
+import dpath
 
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.terraform.checks.resource.base_resource_value_check import BaseResourceCheck
 
-class AKSLoggingEnabled(BaseResourceValueCheck):
+
+class AKSLoggingEnabled(BaseResourceCheck):
     def __init__(self):
         name = "Ensure AKS logging to Azure Monitoring is Configured"
         id = "CKV_AZURE_4"
         supported_resources = ['azurerm_kubernetes_cluster']
         categories = [CheckCategories.KUBERNETES]
         super().__init__(name=name, id=id, categories=categories, supported_resources=supported_resources)
 
-    def get_inspected_key(self):
-        return "addon_profile/[0]/oms_agent/[0]/enabled"
+    def scan_resource_conf(self, conf):
+        two_point_o_path = "addon_profile/[0]/oms_agent/[0]/enabled"
+        three_point_o_path = "oms_agent/[0]"
+        if dpath.search(conf, two_point_o_path) and dpath.get(conf, two_point_o_path)[0]:
+            return CheckResult.PASSED
```

Comment:
Please set `self.evaluated_keys = [two_point_o_path]` (see https://github.com/bridgecrewio/checkov/blob/master/checkov/terraform/checks/resource/aws/CloudfrontDistributionEncryption.py for reference)

Reply:
Thanks for the tip! Right way!